caas/kubernetes/provider/resources/events.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package resources
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/kubernetes"
+)
+
+// ListEventsForObject returns all the events for the specified object.
+func ListEventsForObject(ctx context.Context, client kubernetes.Interface,
```

Comment:
This is unbounded right? I wonder if we should consider using a "since" param or X most recent or something. We only really care about the most recent when computing status I think.